internal/grpc/services/gateway/storageprovidercache.go

```diff
@@ -40,6 +43,20 @@ func userKey(ctx context.Context, ref *provider.Reference) string {
 	return u.Id.OpaqueId + "!" + ref.ResourceId.StorageId + "!" + ref.ResourceId.OpaqueId + "!" + ref.Path
 }
 
+// RemoveFromCache removes a reference from the cache
+func RemoveFromCache(cache *ttlcache.Cache, user *userpb.User, res *provider.ResourceId) {
+	remove := user.Id.OpaqueId
+	if res != nil {
+		remove += "!" + res.StorageId + "!" + res.OpaqueId
+	}
+
+	for _, key := range cache.GetKeys() {
```

Comment:
O(n)? It might be a niche, but couldn't this be O(1)? 🤔

Reply:
Maybe. But I'm not sure. We need to loop over all keys so we can check for prefixes. Any ideas?

Reply:
what he said ;-)

Reply:
not really, I'd argue about using the OpaqueID as keys so it is a 1:1 match, but I think this is totally out of the scope of this. So you are forced to do this gymnastics around that 😓 because I doubt ttlcache is expressive enough to support queries to it.

Reply:
It isn't unfortunately. Removing keys by opaqueid doesn't work, because I need to have the option to remove all keys of a user, so we anyways loop over all keys.